scripts/setup/ubuntu.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Function to print status messages
+print_status() {
+    echo "→ $1"
+}
+
+# Check and install Homebrew if not present
+if ! command_exists curl; then
+    print_status "curl not found. Installing curl..."
+    apt install curl
+    
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to install curl. Please install it manually."
+        exit 1
+    fi
+    print_status "curl installed successfully"
+else
+    print_status "curl is already installed"
+fi
+
+# Check and install Go if not present
+if ! command_exists go; then
+    print_status "Go not found. Installing Go using Homebrew..."
+    apt install golang
```

Comment:
don't think this is the correct command - please refer - https://go.dev/wiki/Ubuntu